### ISSUE TITLE
match_wildcard_for_single_variants should not warn about std types 

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6604]
Lint name: match_wildcard_for_single_variants
note: this is a `pedantic` lint

I tried this code:

```rust
#![warn(clippy::match_wildcard_for_single_variants)]
enum E {
    A(i32),
    B(i32),
}

fn some_e() -> Result<E, E> {
    Err(E::A(2))
}
pub fn main() {
    match some_e() {
        Ok(_) => println!("ok"),
        Err(E::A(numb)) => println!("jackpot: {}", numb),
        _ => panic!("nope"),
    }
}
```

Clippy warns:

````
warning: wildcard match will miss any future added variants
  --> src/main.rs:15:9
   |
15 |         _ => panic!("nope"),
   |         ^ help: try this: `std::prelude::v1::Err(..)`
   |
note: the lint level is defined here
  --> src/main.rs:1:9
   |
1  | #![warn(clippy::match_wildcard_for_single_variants)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_wildcard_for_single_variants

````
which is strange because it would literally break everything if we added more variants to std types like `Result` or `Option`.

### Meta
`clippy 0.1.51 (8a65184 2021-01-16)`

**Comment by camsteffen** [comment]
I think this is a case where the lint is good but the message is wrong. It would be better style IMO to replace `_` with `Err(_)` as it would read "for any other error, do this". If the `_` covers multiple enum variants, then the `_` makes more sense (hence the lint name).
